lib/fieldhand/paginator.rb

```diff
@@ -86,6 +92,24 @@ def parse_response(query = {})
       response_parser
     end
 
+    def retry_request(query = {})
```

Comment:
Rubocop complains about the length of this method; we can cheat by lifting `remaining_retries` into a method argument but it feels a bit too surprising, e.g.

```ruby
def retry_request(query = {}, remaining_retries = retries)
  response = request(query)
  raise ResponseError, response unless response.is_a?(::Net::HTTPSuccess)

  response
rescue ResponseError => e
  raise e unless remaining_retries > 0

  remaining_retries -= 1
  sleep(interval)

  retry
end
```